metacatalog/ext/standard_export/extension.py

```diff
@@ -0,0 +1,127 @@
+from typing import Union
+import os
+
+
+from tqdm import tqdm
+from sqlalchemy.orm import Session
+import xml.etree.ElementTree as ET
+
+
+from metacatalog.ext import MetacatalogExtensionInterface
+from metacatalog.ext.standard_export.util import _parse_iso_information, _init_iso19115_jinja, _validate_xml, _get_uuid
+from metacatalog import api, cmd
+from metacatalog.models import Entry
+from metacatalog.util.results import ImmutableResultSet
+
+
+class StandardsExportExtension(MetacatalogExtensionInterface):
+    r"""
+    Extension to export Entries in standard format.
+    Currently, ISO 19115 export is implemented.
+
+    Adds the method export_iso19115 to :class:`Entry <metacatalog.models.Entry>`
+    which creates the ISO 19115 XML for the :class:`ImmutableResultSet <metacatalog.utils.results.ImmutableResultSet>`
+    of the Entry.
+    The method create_iso19115 is added to the API
+    (metacatalog.api.catalog). This method can be
+    used to export all Entries / ImmutableResultSets
+    in the database session and write the XML files
+    to the folder location specified in ``path``.
+
+    """
+    @classmethod
+    def init_extension(cls):
+        # wrapper which calls StandardsExportExtension.iso19115_export
+        def wrapper_entry(self: Entry, config_dict: dict):  
+            return StandardsExportExtension.iso19115_export(entry_or_resultset=self, config_dict=config_dict)
+        
+        # iso19115_export docstring and name for wrapper function
+        wrapper_entry.__doc__ = StandardsExportExtension.iso19115_export.__doc__
+        wrapper_entry.__name__ = StandardsExportExtension.iso19115_export.__name__
+        
+        # add wrapper to Entry model
+        Entry.export_iso19115 = wrapper_entry
+
+    @classmethod
+    def cli(cls, args):
+        if not args.quiet:
+            print(args.foo.upper())
```

Comment:
This should definitely have a useful name.

Reply:
Sure, I just wanted to try out your example.  
Now that registering a new command works, I can start from here.